i6_models/parts/blstm.py

```diff
@@ -0,0 +1,76 @@
+from dataclasses import dataclass
+import torch
+from torch import nn
+
+from i6_models.config import ModelConfiguration
+
+
+@dataclass()
+class BlstmEncoderConfig(ModelConfiguration):
+    num_layers: int
+    """number of bi-directional LSTM layers, minimum 2"""
+    input_dimension: int
+    """input dimension size"""
+    hidden_dimension: int
+    """hidden dimension of one direction of LSTM, the total output size is twice of this"""
+    dropout: float
+    """nn.LSTM supports an internal Dropout layer"""
+    enforce_sorted: bool = True
+    """
+        keep activated for ONNX-Export, requires that the lengths are sorted decreasing from longest
+
+        Sorting can performed using something like:    
+
+            audio_features_len, indices = torch.sort(audio_features_len, descending=True)
+            audio_features = audio_features[indices, :, :]
+            labels = labels[indices, :]
+            labels_len = labels_len[indices]
```

Comment:
The comment is a bit misleading. Maybe also because `enforce_sorted` in `pack_padded_sequence` is a bit misleading. Actually, what it means:

`enforce_sorted=True`: It does nothing. It expects that the input is sorted. But actually, I think this is also not really true anymore? E.g. see https://github.com/pytorch/pytorch/issues/23079. The question is, what code is really needing this? `pack_padded_sequence` or `LSTM` or sth else?

`enforce_sorted=False`: This will actually automatically sort the sequence internally in `pack_padded_sequence`. So it will make sure (it enforces) that the sequences are sorted. I find the name of this option a bit misleading. The sorting code is basically exactly what you post here, using `torch.sort`.

When you say, `enforce_sorted=True` is needed for ONNX, what this implies is that the sorting code using `torch.sort` is not supported by ONNX? So it means that your code here would also not work. Or vice-versa, if this code works, you could simply also do `enforce_sorted=False`, which does exactly the same.

Reply:
The comment is not misleading at all to me, but I can reformulate it if it is unclear to you.

`keep activated for ONNX-Export` -> you need to set it to true if you want to export this model via ONNX, otherwise it will not work (I tested it)

`requires that the lengths are sorted decreasing from longest` -> if true, you have to care about the sorting yourself, which is also clear and a true statement.

The  example code can be part of the train_step. That way you have a consistent model for training and export, and you also save operations to sort your data only once (for the model and the loss computation) and not twice.

Reply:
As explained, the example code is exactly what happens when you do `enforce_sorted=False`. So, if you say that `enforce_sorted=False` does not work, then your example code also will not work. And that is what I mean by misleading. It sounds like the example code would be a solution here. But it is exactly the code which is used when you do `enforce_sorted=False`, which apparently does not work, as you say.

Reply:
Or do you mean that this example code would not be part of the export?

Reply:
Also, another thing what I mean by misleading: According to https://github.com/pytorch/pytorch/issues/23079, the sorting should not be needed at all.

Reply:
> Or do you mean that this example code would not be part of the export?

Correct, I made the comment more explicit about that.

>  the sorting should not be needed at all.

Feel free to debug and fix it, but I think this is unnecessary effort and so I will not participate in this.